gmso/utils/ff_utils.py

```diff
@@ -91,13 +91,15 @@ def _check_atomtype_existence(tag, ref_dict):
     at3 = tag.attrib.get('type3', tag.attrib.get('class3', None))
     at4 = tag.attrib.get('type4', tag.attrib.get('class4', None))
 
+    atom_classes_set = set(value.atomclass for value in ref_dict.values() if value)
+
     member_types = list(filter(lambda x: x is not None, [at1, at2, at3, at4]))
```

Comment:
everything looks good to me, but i have one question about this line. I know its unrelated to the PR itself, but what exactly does this do?

Looks like the lambda function creates a boolean mask and checks if any of the `at1->at4` are `None`. Im not sure exactly what happens afterwards.

Reply:
A `filter` function takes in a callable and an iterable, and returns a new iterable containing the members of the original iterable for which the callable function evaluates to truthy.

It this case:
1. the lambda function is the callable, [at1, at2, at3, at4] is the iterable. So, it will create a new iterable (casted as a list in the end). So, in this case you could endup with a list containing either 2, 3 or 4, elements, depending on which part of the XML file is being parsed (2 for BondType for example). Basically, the whole filter/list is a syntactic sugar for:
```py
member_types = []
for at in [at1, at2, at3, at4]:
    if at is not None:
        member_types.append(at)
```

Reply:
Oh neat! Makes sense! Thanks for the clarification : )